code/modules/modular_computers/hardware/portable_hard_drive_presets.dm

```diff
@@ -0,0 +1,53 @@
+/obj/item/weapon/computer_hardware/hard_drive/portable/super/preset/all/New()
+    ..()
+    add_programs()
+
+/obj/item/weapon/computer_hardware/hard_drive/portable/super/preset/all/proc/add_programs()
+    for(var/F in typesof(/datum/computer_file/program))
+        var/datum/computer_file/program/prog = new F
+        // Invalid type (shouldn't be possible but just in case), invalid filetype (not executable program) or invalid filename (unset program)
+        if(!prog || !istype(prog) || prog.filename == "UnknownProgram" || prog.filetype != "PRG")
+            continue
+        // Check whether the program should be available for station/antag download, if yes, add it to lists.
+        if(prog.available_on_ntnet)
+            store_file(prog)
+
+
+
+/obj/item/weapon/computer_hardware/hard_drive/portable/backup
+    var/_program = null //Change that far to the file name of the backup program you would like to spawn
+    origin_tech = list() //Nope, no research levels from backup disks
+
+/obj/item/weapon/computer_hardware/hard_drive/portable/backup/New(var/prog_name,var/atom/L)
+    ..()
+    _program = prog_name
+    if(L)
+        loc = L
```

Comment:
Also, unnecessary.

Calling `New(varname)` will automatically spawn the object inside varname. Try nuking this `if(L)` bit and altering line 53 to look like this: ` new /obj/item/weapon/computer_hardware/hard_drive/portable/backup(src, prog.filename)`